conf.d/hydro.fish

```diff
@@ -7,16 +7,25 @@ function $_hydro_git --on-variable $_hydro_git
 end
 
 function _hydro_pwd --on-variable PWD
-    set --local root (command git rev-parse --show-toplevel 2>/dev/null |
-        string replace --all --regex -- "^.*/" "")
-    set --global _hydro_pwd (
-        string replace --ignore-case -- ~ \~ $PWD |
-        string replace -- "/$root/" /:/ |
-        string replace --regex --all -- "(\.?[^/]{1})[^/]*/" \$1/ |
-        string replace -- : "$root" |
-        string replace --regex -- '([^/]+)$' "\x1b[1m\$1\x1b[22m" |
-        string replace --regex --all -- '(?!^/$)/' "\x1b[2m/\x1b[22m"
-    )
+    set --query fish_prompt_pwd_dir_length || set --local fish_prompt_pwd_dir_length 1
+    if test "$fish_prompt_pwd_dir_length" -le 0 || test "$hydro_multiline" = true
+        set --global _hydro_pwd (
+            string replace --ignore-case -- ~ \~ $PWD |
+            string replace --regex -- '([^/]+)$' "\x1b[1m\$1\x1b[22m" |
+            string replace --regex --all -- '(?!^/$)/' "\x1b[2m/\x1b[22m"
+        )
+    else
+        set --local root (command git rev-parse --show-toplevel 2>/dev/null |
+            string replace --all --regex -- "^.*/" "")
+        set --global _hydro_pwd (
+            string replace --ignore-case -- ~ \~ $PWD |
+            string replace -- "/$root/" /:/ |
+            string replace --regex --all -- "(\.?[^/]{"$fish_prompt_pwd_dir_length"})[^/]*/" \$1/ |
+            string replace -- : "$root" |
+            string replace --regex -- '([^/]+)$' "\x1b[1m\$1\x1b[22m" |
+            string replace --regex --all -- '(?!^/$)/' "\x1b[2m/\x1b[22m"
+        )
+    end
```

Comment:
This is the only place that maybe feels a little kludgy and duplicative to me, but I don't think it will have any impact on performance or usability.